src/psyclone/domain/gocean/nodes/gocean_extract_node.py

```diff
@@ -104,8 +112,11 @@ def update_vars_and_postname(self):
 
         '''
         suffix = ""
-        while any(out_var+self._post_name+str(suffix) in self._input_list
-                  for out_var in self._output_list):
+        # The signatures in the input/output list need to be converted
+        # back to strings to easily append the suffix
+        input_string = [str(input_var) for input_var in self._input_list]
```

Comment:
I don't think we can. The main focus here is to create unique names in the output file (NetCDF files). For example, if there is an output variable `a` and a user-named input variable `a_post`. By default, the code would then to create two variables to the NetCDF file, both names `a_post`. This code is to avoid this and create a variable name `a_post1` for the output variable (by appending the '1' to the suffix).
We could use a dummy symbol table here, but I don't see an easy way of generating a single suffix that can be added to any variable, which means we would then need to handle this symbol table back to the code creator for writing and reading the variables, which seems a lot more work then this one loop.

I admit that I want to revisit this to find a better solution, e.g. check if there is a symbol I could use as a separator that cannot be part of a valid Fortran name ("&", "|", "$", ...). Then we would create `a&post` for the output variable, and `a_post` for the input variable. I intend to tackle this as part of #1288 (I thought we had a ticket to use PSyIR for the driver, couldn't find it now). The symbol table should then be able to help creating unique local variable names in the driver (which will also be part of #1288).

Reply:
Oh yes, I see now that it's only creating the suffix. As you say, something to look at later.